blog/2025-04-14-local-dev-pipelines/index.mdx

```diff
@@ -0,0 +1,395 @@
+---
+slug: local-dev-pipelines
+title: "Speeding up local dev: fast feedback when building pipelines"
+description: "Learn how to speed up local development with fast feedback loops when building pipelines."
+authors: [aslan, derik]
+tags: [kratix, promises, pipelines]
+---
+
+Are you building Pipelines and unsure about the best way to test and iterate over them? Worry not. Your questions are about to be answered!
+
+In this blog post, you will gain insights into how you can get faster feedback on your Pipeline development by:
+
+- Running the pipeline stage locally
+- Running automated tests
+- Testing imperative actions
+- Running the stage as a container
+
+We will also look into how to more effectively run system-level tests by
+
+- Populating the image cache
+- Setting the image pull policy
+- Automating end-to-end (e2e) tests
+
+Click "read more" below to continue!
+
+{/* truncate */}
+
+---
+
+## Testing your Pipeline stages
+
+As you may know, a Promise includes a set of workflows that are executed as part of a request for that Promise. The workflow itself is a series of Pipelines, and each Pipeline contains multiple Stages. These stages are usually encapsulated in a Container image.
+
+At the start of a Pipeline Stage, Kratix will provide, in the `/kratix/input` directory, a YAML file (`object.yaml`) representing the user's resource request. This file is the same object that the user has applied to the Platform cluster.
```

Comment:
This is obviously very resource pipeline oriented. Do we need to be clear about promise pipelines as having the same setup but with the promise yaml?